server/extension/extension.go

```diff
@@ -12,24 +13,110 @@ import (
 	"strings"
 	"time"
 
-	applicationpkg "github.com/argoproj/argo-cd/v2/pkg/apiclient/application"
 	v1alpha1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
+	applisters "github.com/argoproj/argo-cd/v2/pkg/client/listers/application/v1alpha1"
+	"github.com/argoproj/argo-cd/v2/server/rbacpolicy"
+	"github.com/argoproj/argo-cd/v2/util/argo"
+	"github.com/argoproj/argo-cd/v2/util/db"
+	"github.com/argoproj/argo-cd/v2/util/security"
 	"github.com/argoproj/argo-cd/v2/util/settings"
 	"github.com/ghodss/yaml"
 	"github.com/gorilla/mux"
 	log "github.com/sirupsen/logrus"
-	"k8s.io/utils/pointer"
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
 const (
 	URLPrefix                    = "/extensions"
-	HeaderArgoCDApplicationName  = "Argocd-Application-Name"
 	DefaultConnectionTimeout     = 2 * time.Second
 	DefaultKeepAlive             = 15 * time.Second
 	DefaultIdleConnectionTimeout = 60 * time.Second
 	DefaultMaxIdleConnections    = 30
+
+	// HeaderArgoCDApplicationName defines the name of the
+	// expected application header to be passed to the extension
+	// handler. The header value must follow the format:
+	//     "<namespace>:<app-name>"
+	// Example:
+	//     Argocd-Application-Name: "namespace:app-name"
+	HeaderArgoCDApplicationName = "Argocd-Application-Name"
+
+	// HeaderArgoCDProjectName defines the name of the expected
+	// project header to be passed to the extension handler.
+	// Example:
+	//     Argocd-Project-Name: "default"
+	HeaderArgoCDProjectName = "Argocd-Project-Name"
 )
 
+// RequestResources defines the authorization scope for
+// an incoming request to a given extension. This struct
+// is populated from pre-defined Argo CD headers.
+type RequestResources struct {
+	ApplicationName      string
+	ApplicationNamespace string
+	ProjectName          string
+}
+
+// Resource defines the Kubernetes resource used for checking
+// the authorization scope.
+type Resource struct {
```

Comment:
Is this struct still used? If not, we should remove it along with the reference to `Argocd-Resource-GVK-Name` below.

Reply:
fixed. good catch